enhancements/cli-in-cluster-management.md

```diff
@@ -0,0 +1,170 @@
+cli-artifacts and oc download links that do not require anything outside the cluster.  The goal is to provide the same for other
+binaries/tools/CLIs.  A local image registry with mirrored CLI images can provide binaries in disconnected clusters.      
+
+The CLI in-cluster manager will provide a mechanism to provide, list, install, and upgrade OpenShift tools/plugins/CLIs from
+within a cluster.  The [ConsoleCLIDownload CRD](https://github.com/openshift/api/blob/master/console/v1/0000_10_consoleclidownload.crd.yaml) 
+will be extended to meet the requirements of a new CLI manager.  Each CLI will provide metadata, descriptions, and image references in a Custom Resource.  
+
+An oc subcommand, `oc cli-index`,  will allow users to interact with CLI CRs to list, update, 
+and gather information about CLIs available to a particular cluster.  The logic for `oc image extract` will
+be extended and modified to include available CLIs (any CLI opts in by supplying the CLI CR).
+
+## Motivation
+
+As more services are created on top of OpenShift, more CLIs are introduced to simplify interaction with these services.
+Some current examples are oc, kubectl, odo, istio, tekton, and knative.  It is difficult for users to discover what tools exist, 
+where to download them from and which version they should download.  We need to simplify as much as possible the interaction
+of services on OpenShift.  We need a mechanism for providing and consuming tools that is simple to add on to as new CLIs are 
+developed from a variety of sources - and this needs to be managed from within a cluster and with disconnected installs.
+This is the motivation for the CLI in-cluster manager.
+
+### Goals
+
+Each component wishing to provide customers with their binaries in-cluster will build and publish artifacts via an official channel to a central index
+from which the CLI in-cluster manager can access them. 
+
+Possible routes for supplying CLI binaries:    
+1.  Central index that has links to where artifacts are stored
+2.  Central repository where all artifacts are stored - [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/)
+currently is where crc, oc, ocp-dev-preview, ocp, odo, and serverless artifacts are published.  
+3.  Images for each CLI. The CLI-manager will provide an extract mechanism, similar to how we currently 
+`oc adm release extract --command oc` and `oc image extract` for each CLI image, and a CR for each CLI will provide the CLI-manager information about each CLI image and also will provide the description of the CLI provided by that image.
+
+Regardless of the index scenario above, CLI-manager will be given the location of each CLI's artifacts via the CLI's CR.    
+
+Option 2 plus Option 3 is the option we as a team are leaning toward.  It requires that each CLI publish artifacts to [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/)
```

Comment:
I prefer option 3. It seems fine to let them publish somewhere publicly downloadable as well, but I'd not connect it to cli manager.

Reply:
updated